drf_spectacular/plumbing.py

```diff
@@ -657,11 +658,21 @@ def modify_for_versioning(patterns, method, path, view, requested_version):
         except Resolver404:
             error(f"namespace versioning path resolution failed for {path}. path will be ignored.")
     elif issubclass(view.versioning_class, versioning.AcceptHeaderVersioning):
-        renderer = view.get_renderers()[0]
-        view.request.META['HTTP_ACCEPT'] = f'{renderer.media_type}; version={requested_version}'
-
-        negotiated = view.perform_content_negotiation(view.request)
-        view.request.accepted_renderer, view.request.accepted_media_type = negotiated
+        # Append the version into request accepted_media_type.
+        # e.g "application/json; version=1.0"
+        # To allow the AcceptHeaderVersioning negotiator going through.
+        if not hasattr(view.request, 'accepted_renderer'):
+            # Probably a mock request, content negotation was not performed, so, we do it now.
+            negotiated = view.perform_content_negotiation(view.request)
+            view.request.accepted_renderer, view.request.accepted_media_type = negotiated
+        try:
+            extracted_version, _ = view.determine_version(view.request, **view.kwargs)
+        except exceptions.NotAcceptable:
+            # A version was found, but did not match the version expected for the schema rendering.
+            # We will override it, in order to force generation of schema to be generated for requested version.
+            extracted_version = None
```

Comment:
sry to be so nitpicky, but i believe this has a new problem. when `NotAcceptable` is thrown and there was indeed an invalid version given in the request, this would mean the `if` below is executed and would for example lead to setting `application/json; version=WRONG; version=1` again. (this may or may not work depending on how the param dict is constructed.)

i would suggest unconditional overriding like this (which is what i meant in my previous comment):
```python
        media_type = _MediaType(view.request.accepted_media_type)
        view.request.accepted_media_type = (
            f'{media_type.full_type}; {view.versioning_class.version_param}={requested_version}'
        )
```
instead of the `try: determine_version...` and `if extracted_version...` (668-675). would you agree here or am i missing something? that stuff is tricky.

Reply:
You are right of being nitpicky. I didn't fully captured the requirements yet. So thank you , your comments are important.
You are right we would end up with `application/json; version=WRONG; version=1` which undesirable.
And your suggestion works better.